src/main/java/at/hannibal2/skyhanni/features/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,471 @@
+                    add("§6Current XP: §b${currentXp.addSeparators()}")
+                    add("§6Needed XP: §b${currentXpMax.addSeparators()}")
+                    add("§6Total XP: §b${totalXp.addSeparators()}")
+                }
+                val nameColor = if (skill == activeSkill) "§e" else "§6"
+                Renderable.hoverTips(buildString {
+                    append("$nameColor${skill.displayName} $level ")
+                    append("§7(")
+                    append("§b${currentXp.addSeparators()}")
+                    if (currentXpMax != 0L) {
+                        append("§6/")
+                        append("§b${currentXpMax.addSeparators()}")
+                    }
+                    append("§7)")
+                }, tips)
+            }
+        }
+    }
+
+    private fun drawETADisplay() = buildList {
+        val activeSkill = activeSkill ?: return@buildList
+        val skillInfo = SkillAPI.storage?.get(activeSkill) ?: return@buildList
+        val xpInfo = skillXPInfoMap[activeSkill] ?: return@buildList
+        val skillInfoLast = oldSkillInfoMap[activeSkill] ?: return@buildList
+        oldSkillInfoMap[activeSkill] = skillInfo
+        val level = if (config.overflowConfig.enableInEtaDisplay.get() || config.customGoalConfig.enableInETADisplay) skillInfo.overflowLevel else skillInfo.level
+
+        val useCustomGoalLevel = skillInfo.customGoalLevel != 0 && skillInfo.customGoalLevel > skillInfo.overflowLevel && customGoalConfig.enableInETADisplay
+        var targetLevel = if (useCustomGoalLevel) skillInfo.customGoalLevel else level + 1
+        if (targetLevel <= level || targetLevel > 400) targetLevel = (level + 1)
+        val currentLevelNeededXp = SkillUtil.xpRequiredForLevel(level.toDouble()) + skillInfo.overflowCurrentXp
+        val targetNeededXp = SkillUtil.xpRequiredForLevel(targetLevel.toDouble())
+        var remaining = if (useCustomGoalLevel) targetNeededXp - currentLevelNeededXp else skillInfo.overflowCurrentXpMax - skillInfo.overflowCurrentXp
+
+        if (!useCustomGoalLevel) {
+            if (skillInfo.overflowCurrentXpMax == skillInfoLast.overflowCurrentXpMax) {
+                remaining = interpolate(remaining.toFloat(), (skillInfoLast.overflowCurrentXpMax - skillInfoLast.overflowCurrentXp).toFloat(), lastGainUpdate.toMillis()).toLong()
+            }
+        }
+
+        add(Renderable.string(buildString {
+            append("§6Skill: §7${activeSkill.displayName} ")
+            if (useCustomGoalLevel && targetLevel != 0) append("§8$level➜§3${targetLevel}") else append(level)
+        }))
+        if (useCustomGoalLevel)
+            add(Renderable.string("§7Needed XP: §e${remaining.addSeparators()}"))
+
+        var xpInterp = xpInfo.xpGainHour
+        if (xpInfo.xpGainHour < 1000) {
+            add(Renderable.string("§7In §cN/A"))
+        } else {
+            add(Renderable.string("§7In §b${Utils.prettyTime((remaining) * 1000 * 60 * 60 / xpInterp.toLong())} " +
```

Comment:
Don't use `Utils.prettyTime`. Use Kotlin Duration, and Duration.format() from SkyHanni's TimeUtils.

After this internal change, you can easily implement this visual change:
Show the time remaining in at least 2 units. I really want to see minutes, not only hours, here. (or minutes and seconds, or days and hours for very large goals)
![grafik](https://github.com/hannibal002/SkyHanni/assets/24389977/c23b7423-d19d-496c-936a-17987c8de506)

Reply:
Show days, hour, minute and seconds.

I'll add a config option for people to choose if they ask for it